ddtrace/contrib/asgi/middleware.py

```diff
@@ -128,12 +128,21 @@ async def __call__(self, scope, receive, send):
         if sample_rate is not None:
             span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, sample_rate)
 
+        host_header = None
+        for key, value in scope["headers"]:
+            if key == b"host":
+                host_header = value.decode("latin-1")
+                break
+
         method = scope.get("method")
         server = scope.get("server")
-        if server and len(server) == 2:
+        scheme = scope.get("scheme", "http")
+        full_path = scope.get("root_path", "") + scope.get("path", "")
+        if host_header is not None:
+            url = f"{scheme}://{host_header}{full_path}"
+        elif server and len(server) == 2:
             port = server[1]
             server_host = server[0] + (":" + str(port) if port is not None and port != 80 else "")
```

Comment:
```suggestion
           default_port = {"http": 80, "https": 443, "ws": 80, "wss": 443}[scheme]       
           server_host = server[0] + (":" + str(port) if port is not None and port != default_port else "")
```

Reply:
The default value for scheme is http (line 139). So that might be a bit redundant. I'll move the dict out

Reply:
well, it is less about the default vs getting a new value we don't know about...... but I am pretty sure it will be a long time or an odd case if that ever happens.